community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelStatement.java

```diff
@@ -58,6 +59,11 @@ public KernelStatement( KernelTransactionImplementation transaction,
     @Override
     public ReadOperations readOperations()
     {
+        if( !transaction.mode().allowsReads() )
```

Comment:
personally i would prefer to have a bit different API, like `transaction.mode().verifyAllowReads()` which will throw exception in case if current mode do not allow reads; that will allow hide all those exceptions and check login inside of access mode and will keep KernelStatement more clear.